### PR TITLE
OCM-8602 | ci: Use Terraform workspaces

### PR DESCRIPTION
**Pull request by radtriste**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**
https://issues.redhat.com/browse/OCM-8602

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

**Comment by radtriste** [comment]
shared VPC issue is known and should be fixed in https://github.com/terraform-redhat/terraform-provider-rhcs/pull/781

**Comment by radtriste** [comment]
`rosa-sts-private-optional-presubmit` is a teardown issue, not related to this PR

File: tests/e2e/cluster_upgrade_test.go [hunk 1]
```diff
@@ -86,45 +90,53 @@ var _ = Describe("Upgrade", func() {
...
+					err = openshift.WaitForOperatorsToBeReady(cms.RHCSConnection, clusterID, timeout)
 					Expect(err).ToNot(HaveOccurred())
 				}
 			})
 
 		It("on Y-stream - [id:63152]", ci.Upgrade,
 			func() {
-
-				if profile.VersionPattern != "y-1" {
+				profile := profile
```
Comment (aaraj7) [inline tests/e2e/cluster_upgrade_test.go:115]: why this line is needed?

Comment (radtriste) [reply tests/e2e/cluster_upgrade_test.go:115]: I think a replace all did that :D nice catch !

**Pull request merged**